src/aind_codeocean_utils/models/config.py

```diff
@@ -0,0 +1,161 @@
+"""Models for config files."""
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel, Field, model_validator
+
+
+class RegisterDataConfig(BaseModel):
+    """
+    Settings for registering data
+    """
+
+    asset_name: str = Field(
+        ..., description="The name to give the data asset."
+    )
+    mount: str = Field(..., description="The mount folder name.")
+    bucket: str = Field(
+        ..., description="The s3 bucket the data asset is located."
+    )
+    prefix: str = Field(
+        ..., description="The s3 prefix where the data asset is located."
+    )
+    public: bool = Field(
+        default=False, description="Whether the data asset is public or not."
+    )
+    keep_on_external_storage: bool = Field(
+        default=True,
+        description="Whether to keep the data asset on external storage.",
+    )
+    tags: Optional[List[str]] = Field(
+        default=None, description="The tags to use to describe the data asset."
```

Comment:
Given the model validator later, the default can just be set to [] and the Optional tag can be removed